lib/fb-user-filestore-client.js

```diff
@@ -133,48 +162,67 @@ class FBUserfilestoreClient extends FBJWTClient {
       delete policy.allowed_types
     }
 
-    /* eslint-disable camelcase */
-    const encrypted_user_id_and_token = this.encryptUserIdAndToken(userId, userToken)
-    /* eslint-enable camelcase */
+    const encrypted_user_id_and_token = this.encryptUserIdAndToken(userId, userToken) // eslint-disable-line camelcase
 
-    const data = {
-      encrypted_user_id_and_token,
-      file,
-      policy
-    }
+    const context = {serviceSlug, userId}
+    const payload = {encrypted_user_id_and_token, file, policy}
 
-    return this.sendPost(urlPattern, {serviceSlug, userId}, data)
-      .then(result => {
-        // useless if no fingerprint returned
-        if (!result.fingerprint) {
-          this.throwRequestError(500, 'ENOFINGERPRINT')
-        }
-        return result
-      })
+    const result = await this.sendPost({url, context, payload}, logger)
+    // useless if no fingerprint returned
+    if (!result.fingerprint) {
+      this.throwRequestError(500, 'ENOFINGERPRINT')
+    }
+    return result
   }
 
   /**
    * Store user file from a file path
    *
-   * @param {string} userId
+   * @param {string} filePath
+   * Path to user file
+   *
+   * @param {object} args
+   * Store args
+   *
+   * @param {string} args.userId
    * User ID
    *
-   * @param {string} userToken
+   * @param {string} args.userToken
    * User token
    *
-   * @param {string} filePath
-   * Path to user file
-   *
-   * @param {object} policy
+   * @param {object} args.policy
    * Policy to apply to file - see store method for details
    *
+   * @param {object} logger
+   * Bunyan logger instance
+   *
    * @return {promise<undefined>}
    *
    **/
-  storeFromPath (userId, userToken, filePath, policy) {
-    return readFile(filePath)
-      .then(file => this.store(userId, userToken, file, policy))
+  async storeFromPath (filePath, args, logger) {
+    const file = await readFile(filePath)
+    args.file = file
+    return this.store(args, logger)
+  }
+}
+
+FBUserFileStoreClient.offline = () => {
+  class FBUserFileStoreClientOffline extends FBUserFileStoreClient {
```

Comment:
Interesting

Reply:
How do you see this tying into to a user journey?

Reply:
Better than doing it in a happenstance way in the runner, having to create stubs for all the methods that will be invoked. I think, anyway...